server/src/test/java/org/cloudfoundry/identity/uaa/db/RemoveAddNewFromProviderConfig_V3_6_0_Tests.java

```diff
@@ -0,0 +1,123 @@
+package org.cloudfoundry.identity.uaa.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.UUID;
+
+import org.cloudfoundry.identity.uaa.constants.OriginKeys;
+import org.cloudfoundry.identity.uaa.provider.IdentityProvider;
+import org.cloudfoundry.identity.uaa.provider.IdpAlreadyExistsException;
+import org.cloudfoundry.identity.uaa.provider.JdbcIdentityProviderProvisioning;
+import org.cloudfoundry.identity.uaa.provider.SamlIdentityProviderDefinition;
+import org.cloudfoundry.identity.uaa.test.JdbcTestBase;
+import org.cloudfoundry.identity.uaa.zone.IdentityZone;
+import org.cloudfoundry.identity.uaa.zone.MultitenancyFixture;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+
+public class RemoveAddNewFromProviderConfig_V3_6_0_Tests extends JdbcTestBase {
+
+    private JdbcIdentityProviderProvisioning provisioning;
+    private RemoveAddNewFromProviderConfig_V3_6_0 migration;
+
+    public static final String SAML_IDENTITY_PROVIDER_WITH_DEPRECATED_PROPERTY_COUNT = "select count(*) from identity_provider where type='"
+            + OriginKeys.SAML + "' and config like '%addNew%'";
+    public static final String SAML_IDENTITY_PROVIDER_TOTAL_COUNT = "select count(*) from identity_provider where type='"
+            + OriginKeys.SAML + "'";
+
+    @Before
+    public void setUpMigration() {
+        this.provisioning = new JdbcIdentityProviderProvisioning(this.jdbcTemplate);
+        this.migration = new RemoveAddNewFromProviderConfig_V3_6_0();
+    }
+
+    @Test
+    public void test_migration() throws Exception {
+        String zoneId = IdentityZone.getUaa().getId();
+
+        // load Database
+        // with Config with addNew Flag only
+        String depricatedPropertyOnlyConfig = "{\"metaDataLocation\":\"mymetadata\",\"idpEntityAlias\":\"1D1DCE7769624FC0B8B8C11344EA67B8\","
+                + "\"zoneId\":\"apm\",\"nameID\":\"unspecified\",\"assertionConsumerIndex\":0,"
+                + "\"metadataTrustCheck\":false,\"showSamlLink\":false,\"socketFactoryClassName\":"
+                + "\"\",\"linkText\":\"Login with GE SSO staging\",\"iconUrl\":null,\"addNew\":false}";
+        this.provisionIdentityProvider(zoneId, "depricatedPropertyOnlyIdp", depricatedPropertyOnlyConfig);
+
+        // load Database
+        // with Config with addNew Flag and AddShadowUserOnLogin Flag
+        String bothPropertiesConfig = "{\"metaDataLocation\":\"mymetadata\",\"idpEntityAlias\":\"1D1DCE7769624FC0B8B8C11344EA67B8\","
```

Comment:
@sanjeevchopra, @ilipkind,

As was discussed previously with Ilya, if values are different, keep the value of addShadowUserOnLogin and remove addNew.  This and other requirements are captured in the user story [US14533](https://rally1.rallydev.com/#/58777512329d/detail/userstory/58795955045)